skimage/morphology/grey.py

```diff
@@ -62,16 +130,15 @@ def erosion(image, selem=None, out=None, shift_x=False, shift_y=False):
            [0, 0, 0, 0, 0]], dtype=uint8)
 
     """
-
-    if image is out:
-        raise NotImplementedError("In-place erosion not supported!")
-    image = img_as_ubyte(image)
-    selem = img_as_ubyte(selem)
-    return cmorph._erode(image, selem, out=out,
-                         shift_x=shift_x, shift_y=shift_y)
+    selem = np.array(selem)
+    selem = _shift_selem(selem, shift_x, shift_y)
+    if out is None:
+        out = np.empty_like(image)
+    nd.grey_erosion(image, footprint=selem, output=out)
+    return out
 
 
-@default_fallback
+@default_selem
 def dilation(image, selem=None, out=None, shift_x=False, shift_y=False):
```

Comment:
@stefanv we need to keep providing this for backwards compatibility. Overall, though, I'd rather deprecate the `shift` keywords entirely and provide some convenience functions for generating the appropriate centered structuring element. e.g.: `square(2)` would be equivalent to `np.array([[0, 0, 0], [0, 1, 1], [0, 1, 1]])`